infrastructure/terraform/modules/activation/main.tf

```diff
@@ -410,26 +437,110 @@ data "external" "ga4_measurement_properties" {
+# This identity is crucial for securely managing secrets and allowing Secret Manager 
+# to interact with other Google Cloud services on your behalf.
+resource "google_project_service_identity" "secretmanager_sa" {
+  provider = google-beta
+  project = null_resource.check_cloudkms_api.id != "" ? module.project_services.project_id : var.project_id
+  service = "secretmanager.googleapis.com"
+}
+ # This Key Ring can then be used to store and manage encryption keys for various purposes, 
+ # such as encrypting data at rest or protecting secrets.
+resource "google_kms_key_ring" "key_ring_regional" {
+  name     = "key_ring_regional-${random_id.random_suffix.hex}"
+  location = var.location
+  project  = null_resource.check_cloudkms_api.id != "" ? module.project_services.project_id : var.project_id
+}
+
+# This key can then be used for various encryption operations, 
+# such as encrypting data before storing it in Google Cloud Storage 
+# or protecting secrets within your application.
+resource "google_kms_crypto_key" "crypto_key_regional" {
+  name     = "crypto-key-${random_id.random_suffix.hex}"
+  key_ring = google_kms_key_ring.key_ring_regional.id
+}
+
+# Defines an IAM policy that explicitly grants the Secret Manager service account 
+# the ability to encrypt and decrypt data using a specific CryptoKey. This is a 
+# common pattern for securely managing secrets, allowing Secret Manager to encrypt 
+# or decrypt data without requiring direct access to the underlying encryption key material.
+data "google_iam_policy" "crypto_key_encrypter_decrypter" {
+  binding {
+    role = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+
+    members = [
+      "serviceAccount:${google_project_service_identity.secretmanager_sa.email}"
+    ]
+  }
+
+  depends_on = [
+    google_project_service_identity.secretmanager_sa,
+    google_kms_key_ring.key_ring_regional,
+    google_kms_crypto_key.crypto_key_regional
+  ]
+}
+
+# It sets the IAM policy for a KMS CryptoKey, specifically granting permissions defined 
+# in another data source.
+resource "google_kms_crypto_key_iam_policy" "crypto_key" {
+  crypto_key_id = google_kms_crypto_key.crypto_key_regional.id
+  policy_data = data.google_iam_policy.crypto_key_encrypter_decrypter.policy_data
+}
+
+# It sets the IAM policy for a KMS Key Ring, granting specific permissions defined 
+# in a data source.
+resource "google_kms_key_ring_iam_policy" "key_ring" {
+  key_ring_id = google_kms_key_ring.key_ring_regional.id
+  policy_data = data.google_iam_policy.crypto_key_encrypter_decrypter.policy_data
+}
+
 # This module stores the values ga4-measurement-id and ga4-measurement-secret in Google Cloud Secret Manager.
 module "secret_manager" {
   source     = "GoogleCloudPlatform/secret-manager/google"
-  version    = "0.4.0"
-  project_id = null_resource.check_secretmanager_api.id != "" ? module.project_services.project_id : var.project_id
+  version    = "~> 0.1"
```

Comment:
It may be best to set the version to a fixed one, to ensure repeatability and be sure that a bug cannot occur when a new version is pushed

Reply:
done